src/main/java/de/hsrm/vegetables/my_food_coop_service/services/StockService.java

```diff
@@ -68,6 +71,44 @@
         return stockRepository.findByStockStatusInAndIsDeleted(stockFilter, !deleteFilter.equals(DeleteFilter.OMIT), sortParameters);
     }
 
+    /**
+     * Returns a page of items currently in stock
+     * deleteFilter controls how deleted entries are treated:
+     * <p>
+     * OMIT: Only elements which haven't been deleted will be included
+     * INCLUDE: Deleted and not deleted items will be returned
+     * ONLY: Only return deleted items
+     *
+     * @param deleteFilter How to treat deleted items
+     * @param offset Pagination offset (first element in returned page)
+     * @param limit Pagination limit (number of elements in returned page)
+     * @return A page of stock items
+     */
+    public Page<StockDto> getStock(DeleteFilter deleteFilter, List<StockStatus> stockFilter, String sortBy, String sortOder, Integer offset, Integer limit) {
+
+        Sort.Direction sortDirection = sortOder.equals("ASC") ? Sort.Direction.ASC : Sort.Direction.DESC;
+        Sort sortParameters = Sort.by(sortDirection, sortByToFieldname(sortBy));
+
+        Pageable pageable = PageRequest.of(offset / limit, limit, sortParameters);
+
+        // No filtering by status
+        if (stockFilter == null || stockFilter.isEmpty()) {
+            return switch (deleteFilter) {
+                case OMIT -> stockRepository.findByIsDeleted(false, pageable);
+                case ONLY -> stockRepository.findByIsDeleted(true, pageable);
+                case INCLUDE -> stockRepository.findAll(pageable);
+            };
+        }
+
+        // No filtering by deleted but by status
+        if (deleteFilter.equals(DeleteFilter.INCLUDE)) {
+            return stockRepository.findByStockStatusIn(stockFilter, pageable);
+        }
+
+        // filtering by stockStatus and deleted
+        return stockRepository.findByStockStatusInAndIsDeleted(stockFilter, !deleteFilter.equals(DeleteFilter.OMIT), pageable);
```

Comment:
Code doubled from the List<> version of this function.

Instead of providing two functions that either return a List<> or a Page<> we  could only provide one function which always provides a Page<>. You can then extract all elements from a Page as a list in cases no pagination is needed.

Have a look at that SO thread: https://stackoverflow.com/questions/42053786/how-to-get-all-results-in-one-page-using-spring-data-pagination

Reply:
Great idea. Followed the most recommended approach of using `Pageable.unpaged()` for unpaged requests in general and the `PageRequest.of(0, Integer.MAX_VALUE)` workaround for unpaged requests with sorting as the first approach does not support sorting.

Reply:
Yeah, was wondering why you used that Pageable.of(0, maxInt) thingy in your code. Not nice but functional and the only way